### ISSUE TITLE
Global Score: Dynamic Updates

### ISSUE THREAD

**Issue by wordpressfan** [wp-media/wp-rocket#7611]
**Context**

Make the Global Score widget update dynamically when individual scores change.

**Dependencies**

Depends on the UI changes #7607

**Acceptance Criteria**

- When adding a new url, refresh the widget to show the `Generating state`.
- When reseting a url, refresh the widget to show the `Generating state`.
- Once a url is completed, the score is updated and show it blurred or not based on credit thing.
- When removing a url, it should update the score
- when removing all urls, we should show the state `No URLs are added`

**Comment by jeawhanlee** [comment]
@wordpressfan This depends on the https://github.com/wp-media/wp-rocket/issues/7608 and it looks tightly coupled
So the same [grooming](https://github.com/wp-media/wp-rocket/issues/7608#issuecomment-3258960611) can be used.

I'm moving this to grooming to review. see grooming in this [issue](https://github.com/wp-media/wp-rocket/issues/7608)

**Comment by wordpressfan** [comment]
No I don't think it's coupled we just need to know the name of the transient mentioned here:
https://github.com/wp-media/wp-rocket/issues/7608#issuecomment-3261748435

and this issue is mainly about the JS and ajax parts.

So here we will mainly trigger js hooks with add/retest/delete and then add js callbacks for them to call the newly created ajax endpoint to get the status from the transient directly.

Conside using WP Rest API as suggested by @remyperona it'll be a great having it here as a seed to move other ajax endpoints.

**Comment by wordpressfan** [comment]
# Grooming

## JS Part
- Adjust this part 
https://github.com/wp-media/wp-rocket/blob/e10785fa7ece044a533f4c48c694cfa0db405ae0/src/js/global/ajax.js#L325

to update the widget HTML with the HTML coming from the response, we need to check with @mostafa-hisham the ID of the widget to be used here.

## PHP Part
- In the [render class](https://github.com/wp-media/wp-rocket/blob/32b4fedc6b16b5cf5e81b701206abb75bd3b9d00/inc/Engine/Admin/PerformanceMonitoring/Render.php#L8) we will have a method to get the HTML for the widget which we can use directly, it'll be implemented in the issue #7607 
- Return a new item [here](https://github.com/wp-media/wp-rocket/blob/e10785fa7ece044a533f4c48c694cfa0db405ae0/inc/Engine/Admin/PerformanceMonitoring/AJAX/Controller.php#L272) that uses the render method to get the widget HTML.

## Effort
[XS]

**Comment by Miraeld** [comment]
Looks good to me

**Comment by wordpressfan** [comment]
I was discussing this with @jeawhanlee 

We have two points here:
1. We can use the Polling system that we have now to update the global score widget.
2. Now we have the global score widget in the dashboard tab but the actions we do with PMA is inside its own tab so we were thinking of only updating the global score widget only when moving from PMA tab into dashboard tab.

this will make is more simple, I'm updating the grooming now to reflect this.

**Comment by jeawhanlee** [comment]
LGTM

**Comment by jeawhanlee** [comment]
All AC will be validated in the parent [issue](https://github.com/wp-media/wp-rocket/issues/7591)
